contracts/erc20-bridge-sampler/test/erc20-bridge-sampler.ts

```diff
@@ -0,0 +1,821 @@
+        });
+
+        it('returns empty for no orders', async () => {
+            const actual = await testContract.queryOrders([]).callAsync();
+            expect(actual).to.deep.eq([]);
+        });
+    });
+
+    describe('queryOrdersAndSampleSells()', () => {
+        const MAKER_TOKEN = randomAddress();
+        const TAKER_TOKEN = randomAddress();
+
+        before(async () => {
+            await testContract.createTokenExchanges([MAKER_TOKEN, TAKER_TOKEN]).awaitTransactionSuccessAsync();
+        });
+
+        it('returns the results of `getOrderInfo()` for each order', async () => {
+            const takerTokenAmounts = getSampleAmounts(TAKER_TOKEN);
+            const orders = createOrders(MAKER_TOKEN, TAKER_TOKEN);
+            const expectedOrderInfos = orders.map(getDeterministicOrderInfo);
+            const [orderInfos] = await testContract
+                .queryOrdersAndSampleSells(orders, SELL_SOURCES.map(n => allSources[n]), takerTokenAmounts)
+                .callAsync();
+            expect(orderInfos).to.deep.eq(expectedOrderInfos);
+        });
+
+        it('can return quotes for all sources', async () => {
+            const sampleAmounts = getSampleAmounts(TAKER_TOKEN);
+            const expectedQuotes = getDeterministicSellQuotes(TAKER_TOKEN, MAKER_TOKEN, SELL_SOURCES, sampleAmounts);
+            const [, quotes] = await testContract
+                .queryOrdersAndSampleSells(
+                    createOrders(MAKER_TOKEN, TAKER_TOKEN),
+                    SELL_SOURCES.map(n => allSources[n]),
+                    sampleAmounts,
+                )
+                .callAsync();
+            expect(quotes).to.deep.eq(expectedQuotes);
+        });
+
+        it('throws if no orders are passed in', async () => {
+            const tx = testContract
+                .queryOrdersAndSampleSells([], SELL_SOURCES.map(n => allSources[n]), getSampleAmounts(TAKER_TOKEN))
+                .callAsync();
+            return expect(tx).to.revertWith(EMPTY_ORDERS_ERROR);
+        });
+
+        it('throws with an unsupported source', async () => {
+            const tx = testContract
+                .queryOrdersAndSampleSells(
+                    createOrders(MAKER_TOKEN, TAKER_TOKEN),
+                    [...SELL_SOURCES.map(n => allSources[n]), randomAddress()],
+                    getSampleAmounts(TAKER_TOKEN),
+                )
+                .callAsync();
+            return expect(tx).to.revertWith(UNSUPPORTED_SOURCE_ERROR);
+        });
+
+        it('throws with non-ERC20 maker asset data', async () => {
+            const tx = testContract
+                .queryOrdersAndSampleSells(
+                    createOrders(MAKER_TOKEN, TAKER_TOKEN).map(o => ({
+                        ...o,
+                        makerAssetData: INVALID_ASSET_PROXY_ASSET_DATA,
+                    })),
+                    [...SELL_SOURCES.map(n => allSources[n]), randomAddress()],
```

Comment:
Nit:
```suggestion
                    SELL_SOURCES.map(n => allSources[n]),
```

This should also be changed in a few places below.